tools/pipeline_perf_test/load_generator/loadgen.py

```diff
@@ -303,6 +328,89 @@ def syslog_worker_thread(self, thread_id: int, args: dict) -> None:
 
         sock.close()
 
+    def syslog_udp_worker_thread(self, thread_id: int, args: dict) -> None:
+        """
+        Worker thread that sends syslog messages to a syslog server via UDP.
+        """
+        syslog_server = os.getenv("SYSLOG_SERVER", "localhost")
+        syslog_port = int(os.getenv("SYSLOG_PORT", "514"))
+
+        print(f"Thread {thread_id}: Using UDP transport to syslog server {syslog_server}:{syslog_port}")
+
+        # Create UDP socket for syslog
+        sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        # TODO: We need to find the right values
+        # sock.setsockopt(socket.SOL_SOCKET, socket.SO_SNDBUF, 1024*1024)
+        sock.setblocking(False)
```

Comment:
I don't think we should set blocking to `False`. This would artificially inflate the failure counts.

I learnt that with blocking behavior, `socket.sendto` there is natural rate limiting through send buffer's back pressure. `socket.sendto` will block until the send buffer has space for it. Moreover, the only work done by this thread is to send UDP messages. With non-blocking behavior, if the send buffer is full, `socket.sendto` would get a `BlockingIOError`. It wouldn't make sense to count that as a failure only to try the same operation again.